packages/flutter/lib/src/cupertino/adaptive_text_selection_toolbar.dart

```diff
@@ -0,0 +1,393 @@
+/// platform with the given children.
+///
+/// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.platforms}
+/// Builds the mobile Cupertino context menu on all mobile platforms, not just
+/// iOS, and builds the desktop Cupertino context menu on all desktop platforms,
+/// not just MacOS. For a widget that builds the native-looking context menu for
+/// all platforms, see [AdaptiveTextSelectionToolbar].
+/// {@endtemplate}
+///
+/// See also:
+///
+/// * [AdaptiveTextSelectionToolbar], which does the same thing as this widget
+///   but for all platforms, not just the Cupertino-styled platforms.
+/// * [CupertinoAdaptiveTextSelectionToolbar.getAdaptiveButtons], which builds
+///   the Cupertino button Widgets for the current platform given
+///   [ContextMenuButtonItem]s.
+class CupertinoAdaptiveTextSelectionToolbar extends StatelessWidget {
+  /// Create an instance of [CupertinoAdaptiveTextSelectionToolbar] with the
+  /// given [children].
+  ///
+  /// See also:
+  ///
+  /// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.buttonItems}
+  /// * [CupertinoAdaptiveTextSelectionToolbar.buttonItems], which takes a list
+  ///   of [ContextMenuButtonItem]s instead of [children] widgets.
+  /// {@endtemplate}
+  /// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.editable}
+  /// * [CupertinoAdaptiveTextSelectionToolbar.editable], which builds the
+  ///   default Cupertino children for an editable field.
+  /// {@endtemplate}
+  /// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.editableText}
+  /// * [CupertinoAdaptiveTextSelectionToolbar.editableText], which builds the
+  ///   default Cupertino children for an [EditableText].
+  /// {@endtemplate}
+  /// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.selectable}
+  /// * [CupertinoAdaptiveTextSelectionToolbar.selectable], which builds the
+  ///   Cupertino children for content that is selectable but not editable.
+  /// {@endtemplate}
+  const CupertinoAdaptiveTextSelectionToolbar({
+    super.key,
+    required this.children,
+    required this.anchors,
+  }) : buttonItems = null;
+
+  /// Create an instance of [CupertinoAdaptiveTextSelectionToolbar] whose
+  /// children will be built from the given [buttonItems].
+  ///
+  /// See also:
+  ///
+  /// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.new}
+  /// * [CupertinoAdaptiveTextSelectionToolbar.new], which takes the children
+  ///   directly as a list of widgets.
+  /// {@endtemplate}
+  /// {@macro flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.editable}
+  /// {@macro flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.editableText}
+  /// {@macro flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.selectable}
+  const CupertinoAdaptiveTextSelectionToolbar.buttonItems({
+    super.key,
+    required this.buttonItems,
+    required this.anchors,
+  }) : children = null;
```

Comment:
Can this just be 
`children = getAdaptiveButtons(buttonItems)`

I noticed you need a context to call the function, but you don't need to if CupertinoTextSelectionToolbarButton.text builds the text in build method instead of the constructor. This will also benefit the inherited rebuild the the text widget only instead of the entire toolbar

Reply:
unless of course you want the const constructor

Reply:
Here you go: https://github.com/flutter/flutter/pull/107193/commits/f8571e42dc06ab088d5734ec4190ea3df681c288

Let me know what you think.  I ended up having to use an `// ignore:` comment.  Let me know if you can think of a better way to do this.

Reply:
I think you can fix the warnning by using
```
required Widget this.child,
```

Reply:
Ah that worked, thanks!